### PR TITLE
Notifications: Add Message-ID header to outgoing emails

### PR DESCRIPTION
**Pull request by janlugt**
Per https://datatracker.ietf.org/doc/html/rfc5322#section-3.6.4, every message should have a "Message-ID:" field. Google recently started enforcing this, see https://support.google.com/a/answer/81126, without a Message-ID, a 
550-5.7.1 error is returned. The gomail library does not automatically add such a header, per the maintainer of that library, it is the responsibility of the library user: https://github.com/go-gomail/gomail/issues/36.

### CODE REVIEW

**Comment by janlugt** [comment]
Hi there, could this get a review? It's basically a oneliner, and it would really help with email notifications to Google Workspace.

**Comment by janlugt** [comment]
It would still be nice if this could be reviewed.

**Comment by janlugt** [comment]
Commenting again to prevent this from getting stale.

**Comment by janlugt** [comment]
Commenting once more...

**Comment by hairyhenderson** [comment]
Hi @janlugt, sorry for the delay here! This fell through the cracks somehow 🤦‍♂️ I'll review now.

File: pkg/services/notifications/smtp.go [hunk 1]
```diff
@@ -93,6 +94,7 @@ func (sc *SmtpClient) buildEmail(ctx context.Context, msg *Message) *gomail.Mess
 	m.SetHeader("From", msg.From)
 	m.SetHeader("To", msg.To...)
 	m.SetHeader("Subject", msg.Subject)
+	m.SetHeader("Message-ID", uuid.New().String())
```
Comment (hairyhenderson) [inline pkg/services/notifications/smtp.go:97]: Easier to just use `uuid.NewString`:

```suggestion
	m.SetHeader("Message-ID", uuid.NewString())
```

Comment (hairyhenderson) [reply pkg/services/notifications/smtp.go:97]: Actually, on reading the RFC, I don't think this is the right format.

The RFC specifies a form like:

```
   msg-id          =   [CFWS] "<" id-left "@" id-right ">" [CFWS]
```

So, I think this should be something like

```suggestion
	m.SetHeader("Message-ID", fmt.Sprintf("<%s@%s>", uuid.NewString(), domain))
```

(where `domain` needs to be extracted from `msg.From`)

Comment (janlugt) [reply pkg/services/notifications/smtp.go]: Thanks for the comments! I've processed them both.

File: pkg/services/notifications/smtp.go [hunk 2]
```diff
@@ -94,6 +95,12 @@ func (sc *SmtpClient) buildEmail(ctx context.Context, msg *Message) *gomail.Mess
 	m.SetHeader("To", msg.To...)
 	m.SetHeader("Subject", msg.Subject)
 
+	at := strings.LastIndex(msg.From, "@")
```
Comment (hairyhenderson) [inline pkg/services/notifications/smtp.go:98]: this won't render correctly for a `From` value of something like: `"Foo Bar" <baz@example.com>`, which is quite a common use - we should probably also cover that case in the unit tests

Comment (janlugt) [reply pkg/services/notifications/smtp.go]: @hairyhenderson, could you take a look at this once more?

**Review by hairyhenderson — APPROVED** [review]
LGTM, thanks for this @janlugt!

**Comment by janlugt** [comment]
> LGTM, thanks for this @janlugt!

Thank you! I'm having some issues with the CI workflows, do you have any idea on how to get those working?

**Comment by janlugt** [comment]
@hairyhenderson, could you help me getting the CI workflows to pass?

**Comment by janlugt** [comment]
@undef1nd, @suntala, could either of you help getting this merged?

**Comment by suntala** [comment]
@janlugt sorry for the delay. The build for the continuous-integration/drone/pr is failing. Looks like you will need to rebase your branch on a newer version of main and force push it.

**Comment by janlugt** [comment]
> @janlugt sorry for the delay. The build for the continuous-integration/drone/pr is failing. Looks like you will need to rebase your branch on a newer version of main and force push it. I'll rerun the job after that.

Thanks! I've just updated my branch.

**Comment by janlugt** [comment]
Thanks for all your help with this!

**Pull request merged**